pre-commit and pre-commit-hooks are not needed to run tests.
https://github.com/python/mypy/issues/15530

--- mypy-1.11.0/test-requirements.in.orig
+++ mypy-1.11.0/test-requirements.in
@@ -8,8 +8,6 @@
 filelock>=3.3.0
 # lxml 4.9.3 switched to manylinux_2_28, the wheel builder still uses manylinux2014
 lxml>=4.9.1,<4.9.3; (python_version<'3.11' or sys_platform!='win32') and python_version<'3.12'
-pre-commit
-pre-commit-hooks==4.5.0
 psutil>=4.0
 pytest>=8.1.0
 pytest-xdist>=1.34.0
